felucca/backend/resource_manager.py

```diff
@@ -1,2 +1,227 @@
+import datetime
+import gridfs
+import pymongo
+from common.job import Job
+from common.task import Task
+from datetime import datetime
+
 class ResourceManager(object):
-    pass
\ No newline at end of file
+    """ResourceManager is responsible for accessing data in database.
+    """
+    
+    def setup(self):
+        """Initialize the connections to the MongoDB server
+        """
+
+        self.__client = pymongo.MongoClient()
+        self.__db = self.__client.felucca
+        self.__jobs_collection = self.__db.jobs
+        self.__tasks_collection = self.__db.tasks
+        self.__fs = gridfs.GridFS(self.__db)
+
+    def __insert_new_task(self, job_id, new_task):
+        """Insert a task into the database
+
+        Args:
+            job_id (ObjectId): The id of the parent job
+            new_task (Task): The task which will be inserted
+        
+        Returns:
+            task_id (ObjectId): The id of the newly inserted task
+        """
+
+        # TODO: Add executable_filename in task (TBD)
+        # TODO: Use string for tool_type? So that we can easily display it to user (TBD)
+
+        # Build the new task
+        new_task_dict = {
+            "tool_type": 0,
+            "command_line_input": new_task.command_line_input,
+            "job_id": job_id,
+            "output_files": {},
+            "log_files": {},
+            "stdout": "",
+            "stderr": "",
+            "is_finished": False,
+            "finished_time": None,
+        }
+
+        # Insert the new task
+        result = self.__tasks_collection.insert_one(new_task_dict)
+        task_id = result.inserted_id
+
+        return task_id
+
+    def insert_new_job(self, new_job):
+        """Insert a job with its tasks into the database
+
+        Args:
+            new_job (Job): The job which will be inserted
+
+        Returns:
+            job_id: The ObjectId of the newly inserted job
+            tasks_id: A list of ids according to the tasks of the job, where the order remains the same
+        """
+
+        self.setup()
+
+        # Build the new job
+        new_job_dict = {
+            "name": new_job.name,
+            "comments": new_job.comments,
+            "created_time": new_job.create_time,
+            "finished_time": None,
+            "is_finished": False,
+        }
+
+        # Insert the new job
+        result = self.__jobs_collection.insert_one(new_job_dict)
+        job_id = result.inserted_id
+
+        # Insert the related tasks
+        tasks_id = []
+        for task in new_job.tasks:
+            task_id = self.__insert_new_task(job_id, task)
+            tasks_id.append(task_id)
+        
+        return job_id, tasks_id
```

Comment:
Cast id from ObjectId to String before returning them.
Also assume all input id is String type.